File: java/utility/MakeContext.java
package voxgig.solardemosdk.utility;

import java.util.Map;

import voxgig.solardemosdk.core.Context;

final class MakeContext {

  private MakeContext() {}

  static Context makeContext(Map<String, Object> ctxmap, Context basectx) {
    return new Context(ctxmap, basectx);
  }
}

File: php/utility/MakeContext.php
<?php
declare(strict_types=1);

// Solardemo SDK utility: make_context

require_once __DIR__ . '/../core/Context.php';

class SolardemoMakeContext
{
    public static function call(array $ctxmap, ?SolardemoContext $basectx): SolardemoContext
    {
        return new SolardemoContext($ctxmap, $basectx);
    }
}
